service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java

```diff
@@ -333,42 +334,26 @@ public OrderWithAddressesResponseDto saveCurrentAddressForOrder(OrderAddressDtoR
                 addressRepo.save(u);
             });
         }
-        Address address;
-        Address forOrderAfterUpdate;
-        if (dtoRequest.getId() != 0) {
-            address = addressRepo.findById(dtoRequest.getId()).orElse(null);
-            forOrderAfterUpdate = modelMapper.map(dtoRequest, Address.class);
-            if (address != null && address.getAddressStatus().equals(AddressStatus.DELETED)) {
-                address = null;
-            }
-        } else {
-            forOrderAfterUpdate = null;
+
+        Address address = addressRepo.findById(dtoRequest.getId()).orElse(null);
+        if (address != null && address.getAddressStatus().equals(AddressStatus.DELETED)) {
```

Comment:
For avoid NPE better to use AddressStatus.DELETED.equals(address.getAddressStatus())